linux/lvm.go

```diff
@@ -324,13 +329,23 @@ func (ls *linuxLVM) CreateLV(vgName string, name string, size uint64,
 		if err := createLVCmd("--zero=y", "--wipesignatures=y", "--size="+sizeB, nameFlag, vgName); err != nil {
 			return nilLV, err
 		}
+
 	case disko.THINPOOL:
 		// When creating a THINPOOL, the name is the thin pool name.
 		if err := createThinPool(name, vgName, size, thinPoolMetaDataSize); err != nil {
 			return nilLV, err
 		}
 	}
 
+	if lvType == disko.THICK || lvType == disko.THIN {
+		if passed, err := checkThenZero(lvPath(vgRealName, name)); err != nil {
```

Comment:
This naming confused me a bit. if `checkThenZero` returns false that means that either the first 4k *or* the last 4k were not zero, right? but we always end up zero in either case (if err !=nil)
so `passed` is just used to write the log?

would it be clearer if we just had a checkzero() function and hoisted the one if at the end of checkthenzero to here?
also what do you think about recording separately which end was not zero?

so it'd be (names are negotiable)
```
if frontWasNonzero, backWasNonzero, err := checkZero(...) err !=nil{
    // handle err
}

if frontWasNonzero || backWasNonzero {
  // write to stderr the values of frontWas and backWas
    withLockedFile(path,
 		func(fp *os.File, fInfo os.FileInfo) error {
 			return zeroStartEnd(fp, int64(0), fileLen, int64(wipeLen))
 		})
}
```

Reply:
> This naming confused me a bit. if `checkThenZero` returns false that means that either the first 4k _or_ the last 4k were not zero, right? but we always end up zero in either case (if err !=nil) so `passed` is just used to write the log?
> 

wasZero starts out as true (assuming it was zeroed by lvm).  Then if the first bits or last bits are not zero, we set 'wasZero=false'.  if wasZero is true (line 412) then we just return true and  do not do the zeroStartEnd call.

so i think its ok as it is.  but your suggestion of putting the logic here is probably nicer.

i had also thought "we're going to wonder which was bad" so your suggestion also covers that. i can adjust.

And yes, 'passed' was just "the check passed".  So we can write that crappy stderr message.

> would it be clearer if we just had a checkzero() function and hoisted the one if at the end of checkthenzero to here? also what do you think about recording separately which end was not zero?
> 
> so it'd be (names are negotiable)
> 
> ```
> if frontWasNonzero, backWasNonzero, err := checkZero(...) err !=nil{
>     // handle err
> }
> 
> if frontWasNonzero || backWasNonzero {
>   // write to stderr the values of frontWas and backWas
>     withLockedFile(path,
>  		func(fp *os.File, fInfo os.FileInfo) error {
>  			return zeroStartEnd(fp, int64(0), fileLen, int64(wipeLen))
>  		})
> }
> ```